src/Modules/UngradedModules.jl

```diff
@@ -483,10 +512,14 @@ function iszero(M::ModuleGens)
 end
 
 function show(io::IO, F::ModuleGens)
-  print(io, "Array of length ", length(F))
-  for i=1:length(F)
-    if isassigned(F.O, i)
-      print(io, "\n", i, " -> ", F.O[i])
+  if F.isGB
+    show_gb(io, F)
+  else
+    print(io, "Array of length ", length(F))
```

Comment:
I understand this is the existing output, but while we are at it, can't we do better? This is not a mere "Array" after all. Why not e.g.
```suggestion
    print(io, "ModuleGens of length ", length(F))
```
or
```suggestion
    print(io, "Module generator array of length ", length(F))
```
or so?